main.js

```diff
@@ -0,0 +1,187 @@
+var posts = [];
+var currentIndex;
+var currentPost;  
+
+//adds new posts based on inputs in main-form and renders them-also stores inputs in posts array with unique id
+var newPost = function () {
+  var $userText = $('#post-text').val();
+  var $userName = $('#name').val(); 
+
+  var post = {
+    id: uuidv4(), 
+    post: $userText,
+    name: $userName,
+    comment: []
+  };
+
+  posts.push(post); 
+
+  //each new post is the last post in the posts array
+  $postText = posts[posts.length-1].post;  
```

Comment:
I don't think these are jQuery objects? Usually we only want to use the `$` in the variable if it is.

Also, this is sort of an anti-pattern - the only reason why these variables are accessible in your render function is because you did not declare them using `var`, which you always want to do. It would be best if you passed them in as arguments when you call `renderPost` below.

Reply:
Thanks, Aaron. I was having the problem where some of my variables weren't being read and I couldn't figure out why. I accidentally didn't declare them and when I noticed and fixed it, I kept getting errors.

Reply:
> I don't think these are jQuery objects? Usually we only want to use the `$` in the variable if it is.
> 
> Also, this is sort of an anti-pattern - the only reason why these variables are accessible in your render function is because you did not declare them using `var`, which you always want to do. It would be best if you passed them in as arguments when you call `renderPost` below.

Thanks, Aaron. I was having the problem where some of my variables weren't being read and I couldn't figure out why. I accidentally didn't declare them and when I noticed and fixed it, I kept getting errors.